lib/modules/datasource/glasskube-packages/index.ts

```diff
@@ -0,0 +1,75 @@
+import { logger } from '../../../logger';
+import { joinUrlParts } from '../../../util/url';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import {
+  GlasskubePackageManifestYaml,
+  GlasskubePackageVersions,
+  GlasskubePackageVersionsYaml,
+} from './schema';
+
+export class GlasskubePackagesDatasource extends Datasource {
+  static readonly id = 'glasskube-packages';
+  static readonly defaultRegistryUrl =
+    'https://packages.dl.glasskube.dev/packages';
+
+  override defaultRegistryUrls = [
+    GlasskubePackagesDatasource.defaultRegistryUrl,
+  ];
+
+  constructor() {
+    super(GlasskubePackagesDatasource.id);
+  }
+
+  override async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    let versions: GlasskubePackageVersions;
+    const result: ReleaseResult = { releases: [] };
+
+    try {
+      const response = await this.http.get(
+        joinUrlParts(registryUrl!, packageName, 'versions.yaml'),
+      );
+      versions = GlasskubePackageVersionsYaml.parse(response.body);
+    } catch (error) {
+      logger.error(
```

Comment:
do we log this as errors in other datasources? 🤔 I think we just pass most errors up.

Reply:
I took the Unity3D datasource as reference and it also logs this kind of error. But I can remove the logging if desired, just let me know.

Reply:
I think at least some errors need to be rethrown, like `ExternalHostError`.

Reply:
use this datasource as a better sample 

https://github.com/renovatebot/renovate/blob/49b1402058400cc2fc8d0c9ebe8cfba5765f781c/lib/modules/datasource/ruby-version/index.ts#L67